src/plugins/workspace/common/types.ts

```diff
@@ -27,3 +27,9 @@ export interface DataSourceConnection {
   description?: string;
   relatedConnections?: DataSourceConnection[];
 }
+
+export enum PermissionModeId {
```

Comment:
Perhaps we can use the `PermissionModeId` from `src/core/types/workspace.ts`?

Reply:
I have trouble to combine them previously, don't remember the reason, but now it works. anyway, combine them into one.